Configuration/Skimming/src/LeptonSkimming.cc

```diff
@@ -0,0 +1,487 @@
+  Trk_container.clear();  MuTracks.clear();  ElTracks.clear(); 
+  object_container.clear(); object_id.clear();  cleanedTracks.clear(); 
+  Epair_ObjectId.clear(); muon_soft.clear(); muon_medium.clear(); muon_tight.clear();
+  Epair_ObjectIndex.clear(); cleanedObjTracks.clear(); cleanedPairTracks.clear();
+    Epair_ObjectIndex.clear(); Epair_ObjectId.clear(); Epair_TrkIndex.clear();
+  //internal stuff
+   ZvertexTrg=-1*std::numeric_limits<float>::max();
+   std::array<float,5> SelectedTrgObj_PtEtaPhiCharge{ {-999,-999,-999,-999,-999}} ;
+   vertex_point.SetCoordinates(-1*std::numeric_limits<float>::max(),-1*std::numeric_limits<float>::max(),-1*std::numeric_limits<float>::max());
+   for (const reco::Vertex& vtx : *vertices) {
+     bool isFake = vtx.isFake();
+     if (isFake) continue;
+     vertex_point.SetCoordinates(vtx.x(),vtx.y(),vtx.z());
+     if (vertex_point.x()!=-1*std::numeric_limits<float>::max()) break;
+    }
+   if (vertex_point.x()==-1*std::numeric_limits<float>::max())
+         return false;
+
+   beam_x= theBeamSpot->x0(); beam_y= theBeamSpot->y0(); beam_z= theBeamSpot->z0();
+  if(!hltFired(iEvent,iSetup,HLTPath_)) 
+    return false;   
+      
+  SelectedTrgObj_PtEtaPhiCharge=hltObject(iEvent,iSetup,HLTFilter_);
+  SelectedMu_DR=std::numeric_limits<float>::max(); 
+  MuTracks.clear();  object_container.clear(); object_id.clear(); nmuons=0;
+  for (const reco::Muon  &mu: *muons){
+    if (fabs(mu.eta())>EtaTrack_Cut) continue;
+    //find triggering muon
+    float deltaRmu=deltaR(mu.eta(),mu.phi(), SelectedTrgObj_PtEtaPhiCharge[1], SelectedTrgObj_PtEtaPhiCharge[2]);
+    if ( deltaRmu<MuTrgMatchCone &&  SelectedMu_DR>deltaRmu ){
+        SelectedMu_DR=deltaR(mu.eta(),mu.phi(), SelectedTrgObj_PtEtaPhiCharge[1], SelectedTrgObj_PtEtaPhiCharge[2]);
+        ZvertexTrg=mu.vz();}
+    //save muons that we want to skim
+    if (!SkimOnlyElectrons){
+      bool tight=false,soft=false;
+      if(vertices.isValid()){
+        tight=isTightMuonCustom(*(&mu),(*vertices)[0]);
+        soft=muon::isSoftMuon(*(&mu),(*vertices)[0]);
+       }    
+       const Track * mutrack= mu.bestTrack();
+       muon_medium.push_back(isMediumMuonCustom(*(&mu)));    
+       muon_tight.push_back(tight); muon_soft.push_back(soft);
+       muon_pt.push_back(mu.pt()); muon_eta.push_back(mu.eta()); muon_phi.push_back(mu.phi()); 
+       auto muTrack=std::make_shared<reco::Track>(*mutrack);
+       MuTracks.push_back(muTrack);
+       object_container.push_back(nmuons);
+       object_id.push_back(13);    
+       nmuons++;}
+    
+  }
+  
+  if (SelectedMu_DR==std::numeric_limits<float>::max() && SkipIfNoMuMatch){
+      return false;
+   }
+  
+
+  //Save electrons we want to skim
+  ElTracks.clear();
+  if (!SkimOnlyMuons){
+    unsigned int count_el=-1;
+   for(const reco::GsfElectron &el : *electrons){  
+      count_el++;
+      bool passConvVeto = !ConversionTools::hasMatchedConversion(*(&el), conversions, theBeamSpot->position());
```

Comment:
@gkaratha this line differs from the master version, where you use "*conversions", why? Here you are passing directly the Handle, I would say that the master version is the correct one